src/main/kotlin/uk/gov/justice/hmpps/offenderevents/services/HMPPSDomainEventsEmitter.kt

```diff
@@ -505,6 +507,46 @@ class HMPPSDomainEventsEmitter(
 
       else -> emptyList()
     }
+
+  private fun VideoAppointmentCancelledEvent.Companion.toDomainEvents(event: VideoAppointmentCancelledEvent): List<HmppsDomainEvent> =
+    event.toDomainEvent().toListOrEmptyWhenNull()
+
+  private val videoAppointmentTypes = listOf("VLB", "VLPM", "VLLA", "VLOO", "VLPA", "VLAP")
+
+  private fun VideoAppointmentCancelledEvent.toDomainEvent(): HmppsDomainEvent? =
+    when {
+      // Only raise an event for the 6 video appointment types, and only if they have been cancelled or deleted
+      (this.recordDeleted || this.scheduleEventStatus == "CANC") && videoAppointmentTypes.contains(this.scheduleEventSubType) -> {
+        prisonApiService.getPrisonerNumberForBookingId(this.bookingId).getOrNull()?.let {
+          HmppsDomainEvent(
+            eventType = "prison-offender-events.video-appointment.cancelled",
+            description = "A video appointment has been cancelled",
+            occurredAt = this.toOccurredAt(),
+            publishedAt = OffsetDateTime.now().toString(),
+            personReference = PersonReference(it),
+          )
+            .withAdditionalInformation("bookingId", this.bookingId)
```

Comment:
do you actually need to copy across all this data too?

Reply:
Not really. I can cut this down to the bare essentials needed.

Reply:
I've cut out 3 attributes not required, but the others still needed by BVL to identify the specific appointment.